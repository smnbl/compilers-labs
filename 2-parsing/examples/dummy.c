int func(int x, int y)
{
    42;

    {
        int z = (42);
    }

    int w;
}

int main()
{
    int array[5];
}
