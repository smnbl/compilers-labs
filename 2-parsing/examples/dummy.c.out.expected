└── Program
    ├── FuncDecl: returnType = 'int', name = 'func'
    │   ├── VarDecl: type = 'int', name = 'x'
    │   ├── VarDecl: type = 'int', name = 'y'
    │   └── CompoundStmt
    │       ├── ExprStmt
    │       │   └── IntLiteral: value = '42'
    │       ├── CompoundStmt
    │       │   └── VarDecl: type = 'int', name = 'z'
    │       │       └── IntLiteral: value = '42'
    │       └── VarDecl: type = 'int', name = 'w'
    └── FuncDecl: returnType = 'int', name = 'main'
        └── CompoundStmt
            └── ArrayDecl: type = 'int', name = 'array'
                └── IntLiteral: value = '5'
